R/XcmsExperiment-functions.R

```diff
@@ -542,6 +542,53 @@
     res[!is.na(res[, "maxo"]), , drop = FALSE]
 }
 
+
+#' Calculates quality metrics for a chromatographic peak. 
+#'
+#' @param x `list` of peak matrices (from a single MS level and from a single
+#'     file/sample).
+#'
+#' @param rt retention time for each peak matrix.
+#'
+#' @param peakArea `matrix` defining the chrom peak area.
+#'
+#' @author Pablo Vangeenderhuysen
+#'
+#' @noRd
+.chrom_peak_beta_metrics <- function(x, rt, peakArea,
+                                           sampleIndex = integer(),
+                                           cn = character(), ...) {
+  res <- matrix(NA_real_, ncol = 2L, nrow = nrow(peakArea))
+  rownames(res) <- rownames(peakArea)
+  colnames(res) <- c("beta_cor","beta_snr")
+  for (i in seq_len(nrow(res))) {
+    rtr <- peakArea[i, c("rtmin", "rtmax")]
+    keep <- which(MsCoreUtils::between(rt, rtr))
+    if (length(keep)) {
+      xsub <- lapply(x[keep], .pmat_filter_mz,
+                     mzr = peakArea[i, c("mzmin", "mzmax")])
+      ## length of xsub is the number of spectra, the number of peaks can
+      ## however be 0 if no peak was found. Maybe we should/need to
+      ## consider adding 0 or NA intensity for those.
+      mat <- do.call(rbind, xsub)
+      if (nrow(mat)) {
+        nr <- vapply(xsub, nrow, NA_integer_)
+        ## can have 0, 1 or x values per rt; repeat rt accordingly
+        rts <- rep(rt[keep], nr)
+        res[i, c("beta_cor", "beta_snr")] <- .get_beta_values(
+          vapply(xsub[nr > 0], sum, NA_real_),
```

Comment:
This line `vapply(xsub[nr > 0], sum, NA_real_)` actually sums each of the numerical matrices in `xsub` (that have at least one row), i.e. it sums the m/z **and** intensity values. Please make sure you only sum the intensity values: `vapply(xsub[nr > 0], function(z) sum(z[, "intensity"]), NA_real_)`.

Could well be that I have the same code in my original function - if so, please correct also there.

Reply:
Also corrected in the original code of `.chrom_peak_intensity_centWave`